tests/core/v5_1/alexandria/test_alexandria_client.py

```diff
@@ -82,3 +91,108 @@ async def _respond_wrong_request_id():
             with pytest.raises(trio.TooSlowError):
                 with trio.fail_after(1):
                     await alice_alexandria_client.ping(bob.node_id, bob.endpoint)
+
+
+@pytest.mark.trio
+async def test_alexandria_client_send_find_nodes(
+    alice, bob, bob_client, alice_alexandria_client
+):
+    async with bob_client.dispatcher.subscribe(TalkRequestMessage) as subscription:
+        await alice_alexandria_client.send_find_nodes(
+            bob.node_id, bob.endpoint, distances=(0, 255, 254),
+        )
+        with trio.fail_after(1):
+            talk_request = await subscription.receive()
+        message = decode_message(talk_request.message.payload)
+        assert isinstance(message, FindNodesMessage)
+        assert message.payload.distances == (0, 255, 254)
+
+
+@pytest.mark.parametrize(
+    "enrs",
+    (
+        ENRFactory.create_batch(1),
+        ENRFactory.create_batch(2),
+        ENRFactory.create_batch(10),
+        ENRFactory.create_batch(50),
+    ),
+)
+@pytest.mark.trio
+async def test_alexandria_client_send_found_nodes(
+    bob, bob_client, enrs, alice_alexandria_client
+):
+    with trio.fail_after(2):
+        async with bob_client.dispatcher.subscribe(TalkResponseMessage) as subscription:
+            await alice_alexandria_client.send_found_nodes(
+                bob.node_id, bob.endpoint, enrs=enrs, request_id=b"\x01\x02",
+            )
+
+            with trio.fail_after(2):
+                first_message = await subscription.receive()
+                decoded_first_message = decode_message(first_message.message.payload)
+                remaining_messages = []
+                for _ in range(decoded_first_message.payload.total - 1):
+                    remaining_messages.append(await subscription.receive())
+
+        all_decoded_messages = (decoded_first_message,) + tuple(
+            decode_message(message.message.payload) for message in remaining_messages
+        )
+        all_received_enrs = tuple(
+            itertools.chain(*(message.payload.enrs for message in all_decoded_messages))
+        )
+        expected_enrs_by_node_id = {enr.node_id: enr for enr in enrs}
+        assert len(expected_enrs_by_node_id) == len(enrs)
```

Comment:
Since they're both sets, adding this assertion might actually make this test *harder* to debug. The test will fail here but won't give any indication of what has gone wrong. Comparing the sets directly, or letting the assert on 147 do all the work, will enable pytest to return better error messages